src/components/annotations/draw.js

```diff
@@ -78,8 +78,12 @@ function drawOne(gd, index) {
 // the plot.
 // axDomainRef: if true and axa defined, draws relative to axis domain,
 // otherwise draws relative to data (if axa defined) or paper (if not).
-function shiftPosition(axa, optAx, dAx, gsDim, vertical, axRef) {
+function shiftPosition(axa, options, dAx, axLetter, gs) {
+    var optAx = options[axLetter];
+    var axRef = options[axLetter + 'ref'];
+    var vertical = axLetter.includes('y');
```

Comment:
Please note that `includes` is not supported in IE.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

Reply:
What about `vertical = /y/.test('axLetter')` ?

Reply:
That should work. Or
```js
vertical = axLetter.indexOf('y') !== -1;
```

Reply:
fixed by 759d37bf9e1eac56cef5d1e4df44e899e20ffadc